client/app/src/main/java/com/microsoft/research/karya/ui/dashboard/DashboardActivity.kt

```diff
@@ -45,7 +50,17 @@ class DashboardActivity : AppCompatActivity() {
         layoutManager = LinearLayoutManager(context)
       }
 
-      syncCv.setOnClickListener { viewModel.fetchNewTasks() }
+      syncCv.setOnClickListener {
+        syncCv.isClickable = false
+        // TODO: START TO SHOW PROGRESS BAR AND THEN FINISH IT
+        try {
+          viewModel.syncWithServer()
+        } catch (e: Exception) {
+          Log.d("DASHBOARD_ACTIVITY", e.toString())
+          // TODO: Add appropriate Error Handling here
+        }
```

Comment:
This should not throw, handle all the errors in viewModel and then we can return `UiState.Error` to the view layer.